main.tf

```
@@ -57,7 +57,7 @@ resource "aws_rds_cluster" "this" {
   port                                = local.port
   db_subnet_group_name                = local.db_subnet_group_name
   vpc_security_group_ids              = compact(concat(aws_security_group.this.*.id, var.vpc_security_group_ids))
-  snapshot_identifier                 = var.snapshot_identifier
+  snapshot_identifier                 = length(keys(var.restore_to_point_in_time)) == 0 ? var.snapshot_identifier : null
```

because of this issue https://github.com/hashicorp/terraform-provider-aws/issues/16454 - lets forgo the changes on line 60 and leave it as:
```suggestion
  snapshot_identifier                 = var.snapshot_identifier
```

this defaults to `null`